### PR TITLE
perf(nuxt): copy assets and prerender in parallel

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
Improve the performance by using [`Promise.all`](https://www.geeksforgeeks.org/javascript-promise-all-method/) syntax :)

- [x] 👌 Enhancement (improving an existing functionality like performance)

_Sorry if I made any mistakes :(_

### CODE REVIEW

**Comment by pi0** [comment]
Thanks for the PR. Do you mind to revert formatting changes

File: packages/nuxt/src/core/nitro.ts [hunk 1]
```diff
@@ -160,19 +160,17 @@ export async function initNitro (nuxt: Nuxt) {
     if (nuxt.options.dev) {
       await build(nitro)
     } else {
-      await prepare(nitro)
```
Comment (pi0) [inline packages/nuxt/src/core/nitro.ts:163]: Prepare needs to be done before both not in parallel.

File: packages/nuxt/src/core/nitro.ts [hunk 2]
```diff
@@ -160,19 +160,17 @@ export async function initNitro (nuxt: Nuxt) {
...
-      await copyPublicAssets(nitro)
-      await prerender(nitro)
+      await Promise.allSettled([prepare(nitro), copyPublicAssets(nitro), prerender(nitro)])
```
Comment (DamianGlowala) [inline packages/nuxt/src/core/nitro.ts:163]: Also, should it be `Promise.all()` instead of `Promise.allSettled()`, as the latter will never reject?

**Comment by sanjaiyan-dev** [comment]
Done @DamianGlowala  & @pi0  👍🏽

File: packages/nuxt/src/core/nitro.ts [hunk 3]
```diff
@@ -161,8 +161,7 @@ export async function initNitro (nuxt: Nuxt) {
       await build(nitro)
     } else {
       await prepare(nitro)
-      await copyPublicAssets(nitro)
-      await prerender(nitro)
+      await Promise.all([copyPublicAssets(nitro), prerender(nitro)])
```
Comment (pi0) [inline packages/nuxt/src/core/nitro.ts:164]: @danielroe Would it break critters? (i assume you somehow depend on build assets being available during prerendering?)

Comment (danielroe) [reply packages/nuxt/src/core/nitro.ts:164]: Nice catch; you are right.

**Comment by pi0** [comment]
Hi @sanjaiyan-dev. I really appreciate time you put on this PR please do it again 💚 (you can make an issue to confirm before making PRs)

Considering build steps, if we do prerendering and copying of assets in parallel, it can cause some issues when prerendering depends on public assets in `/_nuxt/`. We need to read them to inline or optimize some assets. Also it shouldn't be a major bottleneck for performance we could instead focus on increasing prerender parallel performance and prerender build time.

Cheers.

**Comment by sanjaiyan-dev** [comment]
> Hi @sanjaiyan-dev. I really appreciate time you put on this PR please do it again 💚 (you can make an issue to confirm before making PRs)
> 
> Considering build steps, if we do prerendering and copying of assets in parallel, it can cause some issues when prerendering depends on public assets in `/_nuxt/`. We need to read them to inline or optimize some assets. Also it shouldn't be a major bottleneck for performance we could instead focus on increasing prerender parallel performance and prerender build time.
> 
> Cheers.

Ok, tks 💪🙌

**Pull request closed without merging**